contracts/contracts/NoLossDao_v0.sol

```diff
@@ -244,6 +245,7 @@ contract NoLossDao_v0 is Initializable {
   function noLossDeposit(address userAddress)
     external
     depositContractOnly
+    noVoteYet(userAddress) // if the user wants to increase their deposit, they must not have voted already.
```

Comment:
This function is used when users add to their deposit too! The requirement is that they haven't voted in the iteration yet. After they deposit they will no longer be able to vote in that iteration. (similar goes for partial withdrawals - see comment later).

Reply:
Basically, we also need to add this modifier to the partialWithdraw function. I think that will fix the error

Reply:
Actually, I don't think this modifier is necessary. Say you deposit 100DAi and then you vote with your 100 dai, you should be able to top up your deposit even if you have voted? Since you won't be able to vote again that iteration anyway?

Reply:
We should have as little restrictions as possible for money entering the system, this doesn't make sense for me. A user would probably first want to vote before topping their deposit up, otherwise if they top their deposit up, it means they cannot vote that iteration at all which is quite kak